packages/curve-ui-kit/src/features/select-token/ui/modal/TokenList.tsx

```diff
@@ -0,0 +1,163 @@
+import { useState, useMemo, useEffect, memo, type ReactNode } from 'react'
+import Alert from '@mui/material/Alert'
+import AlertTitle from '@mui/material/AlertTitle'
+import Divider from '@mui/material/Divider'
+import MenuList from '@mui/material/MenuList'
+import Stack from '@mui/material/Stack'
+import TextField from '@mui/material/TextField'
+import SearchIcon from '@mui/icons-material/Search'
+
+import { t } from '@ui-kit/lib/i18n'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+
+const { Spacing } = SizesAndSpaces
+
+import type { TokenOption as Option } from '../../types'
+import { FavoriteTokens } from './FavoriteTokens'
+import { TokenOption } from './TokenOption'
+
+// Prevent all the token options from re-rendering if only the balance of a single one has changed.
```

Comment:
This shouldn't be necessary, it should happen automatically :thinking:

Reply:
I tried with and without and didn't notice / measure any difference, so I've removed it.

Reply:
I really need to give virtualization a second try though, when you first open the modal with all tokens the performance is dogshit imo. Not the case when I limit the list to just 5 items. It's also not the re-renders causing the performance hog, checked it with a `console.log(symbol)` inside `TokenOption.tsx`.

Reply:
The comment is still left. Did you check the performance profiler?

Reply:
I've checked the profiler, and could not find any meaningful slowdown in the token list itself. The performance is not bad at all in production, so I chalk it up to a slow laptop on top of local dev mode. It seems the headers and footers with their icons are bigger performance hogs as well. On top of that, after discussion with tutti we're no longer showing the entire token list at once, but will have a "show more" button, as per the new Figma design. Will resolve this for now.